lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -552,21 +532,49 @@ export default class BitcoinProcessor {
     return blockHash;
   }
 
+  /**
+   * Checks if the bitcoin peer has a wallet open for a given address
+   * @param address The bitcoin address to check
+   * @returns true if a wallet exists, false otherwise.
+   */
+  private async walletExists (address: string): Promise<boolean> {
+    console.info(`Checking if bitcoin wallet for ${address} exists`);
+    const request = {
+      method: 'getaddressinfo',
+      params: [
+        address
+      ]
+    };
+
+    const response = await this.rpcCall(request);
+    return response.labels.length > 0 || response.iswatchonly;
```

Comment:
Sanity: under what situation would the response fall on either side of the condition?

Reply:
This was my attempt when comparing two responses from the bitcoin daemon for address info, one of which is being watched while the other is not.
In the bitcoin service, I label the address "sidetree" which makes for an easy check on labels, although really any 'label' would indicate its being watched.
I'm sure theres a number of indicators for if the address is being watched from the addresses, but these were the two most obvious from what this code does.